### PR TITLE
Add array payload configuration to the constructor

### PR DESCRIPTION
**Pull request by jonasemde**
Hi Taylor!

As you mentioned before the payload arguments are now more flexible. This solution is also fully backward compatible.

I hope this code fulfilled your requirements.

TIA
Jonas

### CODE REVIEW

File: src/Slack.php [hunk 1]
```diff
@@ -9,35 +9,48 @@ class Slack
...
      * Create a new Slack instance.
      *
-     * @param  string  $hook
-     * @param  mixed  $channel
-     * @param  string  $message
-     * @return void
+     * @param  string $hook
+     * @param  mixed  $payload
+     * @param  string $message
+     *
```
Comment (GrahamCampbell) [inline src/Slack.php]: please revert all changes to this phpdoc

Comment (GrahamCampbell) [inline src/Slack.php]: apart from the variable rename

File: src/Slack.php [hunk 2]
```diff
@@ -47,11 +60,10 @@ public static function make($hook, $channel = '', $message = null)
      */
     public function send()
     {
-        $message = $this->message ?: ucwords($this->getSystemUser()).' ran the ['.$this->task.'] task.';
-
-        $payload = ['text' => $message, 'channel' => $this->channel];
-
-        Request::post("{$this->hook}")->sendsJson()->body($payload)->send();
+        if(is_null($this->payload['text'])) {
```
Comment (GrahamCampbell) [inline src/Slack.php]: cs

File: src/Slack.php [hunk 3]
```diff
@@ -47,11 +60,10 @@ public static function make($hook, $channel = '', $message = null)
...
+            $this->payload['text'] = ucwords($this->getSystemUser()).' ran the ['.$this->task.'] task.';
+        }
+        Request::post("{$this->hook}")->sendsJson()->body($this->payload)->send();
```
Comment (GrahamCampbell) [inline src/Slack.php]: missing newline

File: src/Slack.php (hunk 2 above)
Comment (GrahamCampbell) [inline src/Slack.php]: also, better to use the `empty` function here

**Comment by taylorotwell** [comment]
No, I meant the payload options should come after the `$message` option, not that they should replace the `$channel` option.

**Pull request closed without merging**